### PR TITLE
tests: round trip test for blob submission/retrieval

### LINKED ISSUE #27: Roundtrip Test for OpenRPC
Create a Roundtrip test to make sure all relevant Methods are working. Make sure the typings are correct and that we don't need fancy serialization and deserialization

### PR DESCRIPTION
**Pull request by tzdybal**
## Overview

Basic round trip test that submits a blob and then reads it back from DA.

Resolves #27.

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

File: client_test.go [hunk 1]
```diff
@@ -105,6 +110,40 @@ func (t *TestSuite) TestClient() {
...
+	t.Zero(txResponse.Code)
+	t.NotZero(txResponse.Height)
+
+	ctx, cancel = context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+	// retrieve data back from DA
+	blobs, err := client.Blob.GetAll(ctx, uint64(txResponse.Height), []share.Namespace{namespace})
+	t.Require().NoError(err)
+	t.Require().NotEmpty(blobs)
+	t.Len(blobs, 1)
```
Comment (tuxcanfly) [inline client_test.go:143]: Should we also check `NotNil(blobs[0])` to avoid a panic.

**Review by tuxcanfly — APPROVED** [review]
OK.

File: client_test.go [hunk 2]
```diff
@@ -105,6 +110,41 @@ func (t *TestSuite) TestClient() {
...
+	namespace, err := share.NewBlobNamespaceV0([]byte{1, 2, 3, 4, 5, 6, 7, 8})
+	t.Require().NoError(err)
+	t.Require().NotEmpty(namespace)
+
+	data := []byte("hello world")
+	blobBlob, err := blob.NewBlobV0(namespace, data)
+	t.Require().NoError(err)
+
+	// write blob to DA
+	txResponse, err := client.State.SubmitPayForBlob(ctx, math.NewInt(100000000), 200000000, []*blob.Blob{blobBlob})
```
Comment (Wondertan) [inline client_test.go:131]: Why not `Blob.Submit`?

**Review by nashqueue — APPROVED** [review]
In a follow-up pr, we should calculate the commitment of the blob and test it against it. That means:
- create commitment 
- get()
- getProof()
- Included(Proof)

This should check all the methods relevant to us

**Comment by nashqueue** [comment]
Another follow-up is to test the submission of multiple blobs. 
1. 2 blobs the same namespace
2. 2 blob different namespaces

File: client_test.go (hunk 2 above)
Comment (tzdybal) [reply client_test.go:131]: Because SubmitPayForBlob is used by https://github.com/rollkit/rollkit/pull/986

**Comment by nashqueue** [comment]
Another test is to ask for a namespace with no blob inside. Basically to catch if the error handling changes.

**Review by gupadhyaya — APPROVED** [review]

File: client_test.go (hunk 2 above)
Comment (gupadhyaya) [reply client_test.go:131]: resolving this conversation to expedite. @Wondertan please create a follow-up, if you need any further change. thanks

**Pull request merged**